tests/Settings/TestSettingVisibilityPresets.py

```diff
@@ -0,0 +1,90 @@
+from unittest.mock import MagicMock
+
+from UM.Preferences import Preferences
+import os.path
+
+from UM.Preferences import Preferences
+from UM.Resources import Resources
+from cura.CuraApplication import CuraApplication
+from cura.Machines.Models.SettingVisibilityPresetsModel import SettingVisibilityPresetsModel
+from cura.Settings.SettingVisibilityPreset import SettingVisibilityPreset
+
+setting_visibility_preset_test_settings = set(["test", "zomg", "derp", "yay", "whoo"])
+
+Resources.addSearchPath(os.path.abspath(os.path.join(os.path.join(os.path.dirname(__file__)), "../..", "resources")))
+Resources.addStorageType(CuraApplication.ResourceTypes.SettingVisibilityPreset, "setting_visibility")
+
+
+def test_settingVisibilityPreset():
```

Comment:
I'd suggest renaming this to something that indicates you're testing the loading from file.